job/job.go

```diff
@@ -34,6 +35,34 @@ type Job struct {
 	instance *packet.Instance
 }
 
+var tinkWorkflowClient = client.GetWorkflowsFromTink
+
+// discoverWorkflowForMac fetches the workflow for a given MAC
+// hwID is the hardware/worker ID corresponding to the MAC
+func discoverActiveWorkflowForMac(hwID string) (bool, error) {
+	var active bool = false
+
+	s := fmt.Sprintf("fetching workflows for hardware %s", hwID)
+	joblog.With("dhcp", hwID).Info(s)
+
+	wcl, err := tinkWorkflowClient(hwID)
+	if err != nil {
+		return false, err
+	}
+
+	// wcl is workflowContextList, for more details refer to workflow.pb.go in tink
+	for _, wf := range wcl.WorkflowContexts {
+		if wf.CurrentActionState == 0 {
+			active = true
+			s := fmt.Sprintf("found active workflow for hardware %s with workflow "+
+				"id %s", hwID, wf.WorkflowId)
+			joblog.With("dhcp", hwID).Info(s)
```

Comment:
If you `return true` here I think you do not need to break the loop. And you can `return false` at the end. Removing the allocation at line 43.

I think I am also not 100% sure about the name/doc block for this function.

is this function returning `true` if the given Hardware has an active workflow? If so can we call this function something like: `isHardwareExecutingWorkflow`? I tend to use `is` as a prefix when a function returns `true/false` because it is a bit more sematic.

Reply:
Agree with all of this

Reply:
done

Reply:
done, renamed it to `isHardwareHavingActiveWorkflow`